src/ui/components/Rating/index.js

```diff
@@ -28,14 +28,23 @@ type InternalProps = {|
   i18n: I18nType,
 |};
 
-export class RatingBase extends React.Component<InternalProps> {
+type StateType = {|
+  hoveringOverStar: number | null,
+|};
+
+export class RatingBase extends React.Component<InternalProps, StateType> {
   static defaultProps = {
     className: '',
     readOnly: false,
     styleSize: 'large',
     yellowStars: false,
   };
 
+  constructor(props: InternalProps) {
+    super(props);
+    this.state = { hoveringOverStar: null };
```

Comment:
I chose `setState` here because I wanted the UI to be snappy when hovering. Switching to `setUIState()` also burdens the caller with having to generate an ID. I dunno, I'm still thinking on it.